packages/vm/src/runBlock.ts

```diff
@@ -407,6 +416,51 @@ async function applyBlock(this: VM, block: Block, opts: RunBlockOpts): Promise<A
   return blockResults
 }
 
+/**
+ * This method runs the logic of EIP 2935 (save blockhashes to state)
+ * It will put the `parentHash` of the block to the storage slot of `block.number - 1` of the history storage contract.
+ * This contract is used to retrieve BLOCKHASHes in EVM if EIP 2935 is activated.
+ * In case that the previous block of `block` is pre-EIP-2935 (so we are on the EIP 2935 fork block), additionally
+ * also add the currently available past blockhashes which are available by BLOCKHASH (so, the past 256 block hashes)
+ * @param this The VM to run on
+ * @param block The current block to save the parent block hash of
+ */
+export async function accumulateParentBlockHash(this: VM, block: Block) {
+  const historyAddress = Address.fromString(
+    bigIntToHex(this.common.param('vm', 'historyStorageAddress'))
+  )
+
+  if ((await this.stateManager.getAccount(historyAddress)) === undefined) {
+    await this.evm.journal.putAccount(historyAddress, new Account())
+  }
+
+  async function putBlockHash(vm: VM, hash: Uint8Array, number: bigint) {
+    const key = setLengthLeft(bigIntToBytes(number), 32)
+    await vm.stateManager.putContractStorage(historyAddress, key, hash)
+  }
+  await putBlockHash(this, block.header.parentHash, block.header.number - BIGINT_1)
+
+  // Check if we are on the fork block
+  const forkTime = this.common.eipTimestamp(2935)
+  if (forkTime === null) {
+    throw new Error('EIP 2935 should be activated by timestamp')
+  }
```

Comment:
I don't think that we need this extra `eipTimestamp()` method. Isn't this the normal `isActivatedEIP()` call??

Also: such a check should likely be at the beginning of the method (?).

Reply:
I have added this because I need to detect we are on the first fork block. The reason is that on the first fork block, we have to store all the `MIN_HISTORY_SERVE_WINDOW` hashes. I don't know if there is another way to detect if we are on the fork block? I think we need to fork timestamp and then the exact condition `if (parentBlock.header.timestamp < forkTime) {` (this only happens once and is on the fork block)

Reply:
Ah, you are right, sorry, please dismiss! 🙂 

Should then the whole method get a check at the beginning if 2935 is activated and throw otherwise?

Reply:
I am comparing it with `accumulateParentBeaconBlockRoot`, this also does not have a check for `common.isActivatedEIP(4788)`, but for consistency I can add (but then I will add it to both methods)

Reply:
Yes, maybe do (for both).